### DISCUSSION TITLE
How to addin specific package if cake running in windows

### DISCUSSION THREAD

**Discussion by soroshsabz** [cake-build/cake#3645] [Q&A]
ITNOA

I have cake file that want to run in Windows and Ubuntu.

In Windows we need Cake.EntityFramework `addin` and in Ubuntu we do not need it.

So I want to write some platform specific `addin` like below

```
#if WINDOWS
#addin nuget:?package=Cake.EntityFramework&version=0.3.0
#endif
```

But I do not know how to do it?

I ask it in [SO](https://stackoverflow.com/q/69767385/1539100) too.

related to #3646

**Comment by nils-a** [comment]
You can't do that in one Cake file, but it should be possible in two: You could check for Windows using `IsRunningOnWindows` and then use `CakeExecuteScript()` to call the windows-specific script.

**Reply by soroshsabz** [reply]
Why? It is cause very redundant logic in multiple files, and make hard to maintain cake files.

**Reply by nils-a** [reply]
I'm sorry - I feel I don't understand the question. Is your question
"Why does `CakeExecuteScript()` exist?" - the answer to that is because some people like to have their files splitted.

**Reply by soroshsabz** [reply]
I want to have one file for `cake` because many of content is same for Windows and Linux, and just few lines is different, so I like to can spilt some part of `cake` file when  this file run in Windows with directive.

**Reply by nils-a** [reply]
> I want to have one file for cake because many of content is same

I understand. But, as `#if` does not work currently with `#addin` I tried to give you an alternative. 

Using two files and "calling" one using `CakeExecuteScript()` was the first solution that came to my mind. Another is using three files: `windows.cake`, `linux.cake` and `shared.cake` and having both `windows.cake` and `linux.cake` reference `shared.cake` via the `#load` directive. But that's also not "one file".

I am afraid I can not offer a solution that does both, fix your problem at hand and adhere to the restraint of being "one file".

EDIT: `CakeExecuteScript()` has an overload that takes a string as the cake script - would that be a way to keep things in "one file"?

**Reply by nick5454** [reply]
I pass in a parameter "IsWindows" to resolve my issue.
